tests/integration/actions/test_list_table_names.py

```diff
@@ -110,5 +108,5 @@ def test_running_list_table_names_action_fails_for_unreachable_datasource(
 
     # Act & Assert
     # Check that the action was unsuccessful and an error was raised.
-    with pytest.raises(ValueError):
+    with pytest.raises(KeyError):
```

Comment:
Both these errors are surprising to see here. Where is the KeyError coming from?